### PR TITLE
Remove unnecessary lodash

### PR DESCRIPTION
**Pull request by TheIronDev**
### Description

Tech debt; Removing some unnecessary references of lodash.

### Code sample

* Before:

```
if(_.get(object, 'param'))
```

* After:

```
if(!!object.param)
```

### CODE REVIEW

**Comment by TheIronDev** [comment]
Requesting a review from @inlined

File: src/lifecycle.ts [hunk 1]
```diff
@@ -72,7 +71,7 @@ export class FirebaseFunctionsTest {
 
   /** Complete clean up tasks. */
   cleanup() {
-    forEach(this._oldEnv, (val, varName) => {
+    Object.entries(this._oldEnv).forEach(([varName, val]) => {
```
Comment (inlined) [inline src/lifecycle.ts:74]: nit: IIRC Google style guide says we should just be doing `for of` instead of `forEach`

File: src/main.ts [hunk 2]
```diff
@@ -109,13 +112,13 @@ export type WrappedScheduledFunction = (
...
 ): WrappedScheduledFunction | WrappedFunction {
-  if (!has(cloudFunction, '__trigger')) {
+  if (!cloudFunction.__trigger) {
     throw new Error(
       'Wrap can only be called on functions written with the firebase-functions SDK.'
     );
   }
 
-  if (get(cloudFunction, '__trigger.labels.deployment-scheduled') === 'true') {
+  if (cloudFunction?.__trigger?.labels && cloudFunction?.__trigger?.labels['deployment-scheduled'] === 'true') {
```
Comment (inlined) [inline src/main.ts:121]: `if (cloudFunction.trigger?.labels?.["deployment-scheduled"] === 'true') {`

File: src/main.ts [hunk 3]
```diff
@@ -133,22 +136,23 @@ export function wrap<T>(
...
+  if (!cloudFunction.run) {
     throw new Error(
       'This library can only be used with functions written with firebase-functions v1.0.0 and above'
     );
   }
 
   const isCallableFunction =
-    get(cloudFunction, '__trigger.labels.deployment-callable') === 'true';
+    cloudFunction?.__trigger?.labels &&
+    cloudFunction?.__trigger?.labels['deployment-callable'] === 'true';
```
Comment (inlined) [inline src/main.ts:155]: Here again you can use `?.[]` to do null coalescing in a dictionary access.

File: src/main.ts [hunk 4]
```diff
@@ -172,14 +176,13 @@ export function wrap<T>(
       const defaultContext = _makeDefaultContext(cloudFunction, options, data);
 
       if (
-        has(defaultContext, 'eventType') &&
-        defaultContext.eventType !== undefined &&
+        !!defaultContext.eventType &&
```
Comment (inlined) [inline src/main.ts:179]: FWIW we don't typically do bool casting. Don't need to remove it though.

**Review by inlined — APPROVED** [review]
LGTM. Are we able to actually remove lodash from our package.json?

File: src/lifecycle.ts (hunk 1 above)
Comment (TheIronDev) [reply src/lifecycle.ts]: Done!

File: src/main.ts (hunk 2 above)
Comment (TheIronDev) [reply src/main.ts]: Done (thanks for calling out the improved syntax!)

File: src/main.ts (hunk 3 above)
Comment (TheIronDev) [reply src/main.ts]: !! Thanks!

File: src/main.ts (hunk 4 above)
Comment (TheIronDev) [reply src/main.ts]: Removed bool casting to match existing code convention.

**Comment by TheIronDev** [comment]
We could *almost* remove lodash here. `firestore.ts` has a few strange things that I was going to tackle in a follow-up pr.

File: src/main.ts [hunk 5]
```diff
@@ -133,22 +136,22 @@ export function wrap<T>(
   }
 
   if (
-    has(cloudFunction, '__trigger.httpsTrigger') &&
-    get(cloudFunction, '__trigger.labels.deployment-callable') !== 'true'
+    !!cloudFunction?.__trigger?.httpsTrigger &&
+    cloudFunction?.__trigger?.labels['deployment-callable'] !== 'true'
```
Comment (inlined) [inline src/main.ts:140]: Your index access needs to be null coalescing here.

**Review by inlined — APPROVED** [review]

File: src/main.ts (hunk 5 above)
Comment (TheIronDev) [reply src/main.ts]: Whups! Thanks for the catch!

**Comment by TheIronDev** [comment]
Closing pull request, there are a bunch of conflicts that would take a bit too long to resolve..

**Pull request closed without merging**